### PR TITLE
[10.x.x] Screen Space Ambient Occlusion (Part 1&2)

### PR DESCRIPTION
**Pull request by ellioman**
### Purpose of this PR
This is the first and second part of an SSAO implementation for URP. 

SSAO will be added incrementally to the master branch. The plan is as follows:
- Part-1: SSAO with Depth only mode.
- Part-2: SSAO with Depth Normals mode, DepthNormal prepass and ConfigureInputs() for ScriptableRenderPasses
- Part-3: SSAO with Deferred mode
- Part-4: SSAO with Volume Framework support.

#### Known issues
- SSAO doesn't work with dynamic resolution as we need to update the _ScreenParams parameter sent to our shaders. This will will be fixed in another PR.

#### This PR adds the following:
- Removes the 8 renderer limit in the URP Asset
- Adds SSAO render feature and editor script for that
- Adds new Shader keyword _SCREEN_SPACE_OCCLUSION
- Adds a virtual ScriptableRenderFeature.Dispose() function which is invoked during ScriptableRenderer.Dispose()
- Adds normalizedScreenSpaceUV parameter to the InputData struct
- Adds a DepthNormal pass in URP Shaders and a Depth Normals Prepass
- Adds ConfigureInput() function to ScriptableRenderPasses
- Makes the ForwardRenderer check inputs set by render passes for input requirements
- Adds Universal Graphics Test scene: 142_SSAO_DepthNormal_Projection
- Adds Universal Graphics Test scene: 143_SSAO_DepthNormal_Orthographic

---

### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-2019.3` label. After you backport the PR, the label changes to `backported-2019.3`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---

### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):

### CODE REVIEW

**Comment by schodemeiss** [comment]
Any idea when this will get merged in? Would be great to get even some basic SSAO support in my URP project.

**Comment by ellioman** [comment]
@schodemeiss Sorry for the late reply.
The current plan is to finish this PR and merge it in to master this week. Hopefully there will be no setbacks. 

If everything goes to plan then we will then backport this to the 8.x.x and 7.x.x branches and release it as part of 8.2 and 7.5.

**Comment by schodemeiss** [comment]
> @schodemeiss Sorry for the late reply.
> The current plan is to finish this PR and merge it in to master this week. Hopefully there will be no setbacks.
> 
> If everything goes to plan then we will then backport this to the 8.x.x and 7.x.x branches and release it as part of 8.2 and 7.5.

No problem at all, was just curious. I can't wait to try it out. I'm currently running the new 2020.2a10. I assume this works on that version of Unity?

**Comment by ellioman** [comment]
@schodemeiss It will yes, once the PR + backports are merged and we release a package containing it. The normal process is to get things in to the latest Unity releases (in this case 2020.2) and then backport to older ones.

File: com.unity.render-pipelines.universal/Runtime/Passes/ScriptableRenderPass.cs [hunk 1]
```diff
@@ -4,6 +4,19 @@
...
+    /// Input requirements for <c>ScriptableRenderPass</c>.
+    /// </summary>
+    /// <seealso cref="ConfigureInput"/>
+    [Flags]
+    public enum ScriptableRenderPassInput
+    {
+        None = 0,
+        Depth = 1,
+        Normal = 2,
+        Color = 4,
```
Comment (pbbastian) [inline com.unity.render-pipelines.universal/Runtime/Passes/ScriptableRenderPass.cs:17]:
```suggestion
        None = 0,
        Depth = 1 << 0,
        Normal = 1 << 1,
        Color = 1 << 2,
```

Comment (ellioman) [reply com.unity.render-pipelines.universal/Runtime/Passes/ScriptableRenderPass.cs]: Done!

**Comment by hybridherbst** [comment]
@ellioman great that this finally landed - would you mind explaining why you removed the backport labels? A lot of people are really hoping that this will land in 7.x as you know :)

**Pull request merged**